src/main/scala/nl.knaw.dans.easy.dd2d/DepositToDataverseMapper.scala

```diff
@@ -89,80 +91,86 @@ class DepositToDataverseMapper(narcisClassification: Elem, isoToDataverseLanguag
 
     // Data vault
-    addVaultValue(dataVaultFields, BAG_ID, vaultMetadata.dataverseBagId)
-    addVaultValue(dataVaultFields, NBN, vaultMetadata.dataverseNbn)
-    addVaultValue(dataVaultFields, DANS_OTHER_ID, vaultMetadata.dataverseOtherId)
-    addVaultValue(dataVaultFields, DANS_OTHER_ID_VERSION, vaultMetadata.dataverseOtherIdVersion)
-    addVaultValue(dataVaultFields, SWORD_TOKEN, vaultMetadata.dataverseSwordToken)
+    addPrimitiveFieldSingleValue(dataVaultFields, BAG_ID, Option(vaultMetadata.dataverseBagId))
+    addPrimitiveFieldSingleValue(dataVaultFields, NBN, Option(vaultMetadata.dataverseNbn))
+    addPrimitiveFieldSingleValue(dataVaultFields, DANS_OTHER_ID, Option(vaultMetadata.dataverseOtherId))
+    addPrimitiveFieldSingleValue(dataVaultFields, DANS_OTHER_ID_VERSION, Option(vaultMetadata.dataverseOtherIdVersion))
+    addPrimitiveFieldSingleValue(dataVaultFields, SWORD_TOKEN, Option(vaultMetadata.dataverseSwordToken))
 
     assembleDataverseDataset()
   }
 
   private def assembleDataverseDataset(): Dataset = {
     val versionMap = mutable.Map[String, MetadataBlock]()
     addMetadataBlock(versionMap, "citation", "Citation Metadata", citationFields)
-    addMetadataBlock(versionMap, "archaeologyMetadata", "Archaeology-Specific Metadata", archaeologySpecificFields)
-    addMetadataBlock(versionMap, "temporal-spatial", "Temporal and Spatial Coverage", temporalSpatialFields)
+    addMetadataBlock(versionMap, "dansArchaeologyMetadata", "Archaeology-Specific Metadata", archaeologySpecificFields)
+    addMetadataBlock(versionMap, "dansTemporalSpatial", "Temporal and Spatial Coverage", temporalSpatialFields)
     addMetadataBlock(versionMap, "dansRights", "Rights Metadata", rightsFields)
-    addMetadataBlock(versionMap, "dataVault", "Data Vault Metadata", dataVaultFields)
+    addMetadataBlock(versionMap, "dansDataVaultMetadata", "Data Vault Metadata", dataVaultFields)
     val datasetVersion = DatasetVersion(metadataBlocks = versionMap.toMap)
     Dataset(datasetVersion)
   }
 
-  private def addPrimitiveFieldSingleValue(metadataBlockFields: ListBuffer[MetadataField], name: String, sourceNodes: NodeSeq, nodeTransformer: Node => Option[String] = AnyElement toText): Unit = {
+  private def addPrimitiveFieldSingleValue(metadataBlockFields: mutable.HashMap[String, AbstractFieldBuilder], name: String, sourceNodes: NodeSeq, nodeTransformer: Node => Option[String] = AnyElement toText): Unit = {
     sourceNodes
       .map(nodeTransformer)
       .filter(_.isDefined)
       .map(_.get)
       .take(1)
-      .foreach(v => metadataBlockFields += PrimitiveSingleValueField(name, v))
+      .foreach(v => {
+        metadataBlockFields.getOrElseUpdate(name, new PrimitiveFieldBuilder(name, multipleValues = false)) match {
+          case b: PrimitiveFieldBuilder => b.addValue(v)
+          case _ => throw new IllegalArgumentException("Trying to add non-primitive value(s) to primitive field")
```

Comment:
Will these Exceptions ever be thrown?

Reply:
It will only be thrown if there is a programming error, i.e. when some line tries to add for example a compound value to a primitive field. It would of course be better to catch that during type-checking but I didn't see a way to do that, so a run-time check is my fallback option. View it as a kind of "assert" check.

So, essentially you are right: this *should* never happen.